public/sw.js

```
@@ -1,33 +1,62 @@
+  };
+}
 
-self.addEventListener('notificationclick', (event) => {
-  event.notification.close();
-  const screen = event.notification.data?.screen;
-  const url = screen ? `/${screen}` : '/';
-  event.waitUntil(
-    clients
-      .matchAll({ type: 'window', includeUncontrolled: true })
-      .then((windowClients) => {
+if (typeof self !== 'undefined' && typeof self.addEventListener === 'function') {
```

ESLint

[error] 'self' is not defined.

(no-undef)